### ISSUE TITLE
Inappropriate mean in loss_functional with rnn

### ISSUE THREAD

**Issue by rarilurelo** [DeepX-inc/machina#118]
When RNN is used, loss is averaged through (timestep, batchsize). However steps after terminate are masked by output_masks. Episode length must be arranged same length for using RNN.
```python:loss_functional.py
pol_loss = torch.mean(pol_loss * out_masks)
```
We have to calculate this like below.
```
timestep = torch.sum(out_masks, dim=0)
pol_loss = torch.sum(pol_loss * out_masks) / (timestep * batchsize)
```

**Comment by rarilurelo** [comment]
#198
